lib/rules/collect-like-terms.js

```diff
@@ -164,4 +166,72 @@ const COLLECT_LIKE_TERMS = defineRule(
     }
 )
 
+export const ADD_POLYNOMIAL_TERMS = defineRule(
+    // MATCH PATTERN
+    (node) => {
+        let hasLikeTerms = false
+
+        if (isPolynomial(node)) {
+
+            // e.g 2x + 2x + 4 + 6
+            // coefficient map: {'x': [[2 node] [2 node]}
+            // constants: [[4 node] [6 node]]
+            const {constants, coefficientMap} = getCoefficientsAndConstants(node)
+
+            // checks if at least one key has more than 1
+            // coefficient term
+            hasLikeTerms = Object.keys(coefficientMap)
+                .some(key => coefficientMap[key].length > 1)
+        }
+        return hasLikeTerms ? {node} : null
+    },
+
+
+    // REWRITE PATTERN
+    (node) => {
+        const {constants, coefficientMap} = getCoefficientsAndConstants(node)
+
+        // One gigantic applyNode
+        // Use example: 2x + 3x + 4
+        const result = build.applyNode(
+            'add',
+
+            // for each identifier, generate the simplified term
+            // e.g 2x + 3x -> 5x
+            Object.keys(coefficientMap).sort().map(key => {
+                // [[2 node], [3 node]]
+                const coeffs = coefficientMap[key]
+                // x
+                const variable = parse(key)
+
+                // 5
+                const newCoeff =
+                      coeffs.reduce((runningTotal, value) =>
+                                    runningTotal + query.getValue(value), 0)
+
+                // TODO: find a better way to specify the location
+                // [[5 node]]
+                const newCoeffNode =
+                      build.numberNode(newCoeff, null, null)
+
+                // [[5x node]]
+                const term = build.applyNode(
+                    'mul', [clone(newCoeffNode), clone(variable)],
+                    null, {implicit: true}
+                )
+
+                return term
```

Comment:
You'll want to flatten this term before you return it so that this will handle cases like `2 x y + 3 x y`